src/message_element_metadata_owner.cc

```diff
@@ -120,37 +125,58 @@ namespace grpc_labview {
             return;
         }    
         int clusterOffset = 0;
+        int maxAlignmentRequirement = 0;
         for (auto element: metadata->_elements)
         {
-            auto elementType = element->type;
-            auto nestedElement = element;
-            while (elementType == LVMessageMetadataType::MessageValue)
-            {
-                auto nestedMetadata = FindMetadata(element->embeddedMessageName);
-                auto nestedElement = nestedMetadata->_elements.front();
-                elementType = nestedElement->type;
-            }    
-            clusterOffset = AlignClusterOffset(clusterOffset, element->type, element->isRepeated);
-            element->clusterOffset = clusterOffset;
             if (element->type == LVMessageMetadataType::MessageValue)
-            {                
+            {
                 auto nestedMetadata = FindMetadata(element->embeddedMessageName);
                 UpdateMetadataClusterLayout(nestedMetadata);
+                int alignmentRequirement = 0;
+                int elementSize = 0;
                 if (element->isRepeated)
-                {                
-                    clusterOffset += ClusterElementSize(element->type, element->isRepeated);
+                {
+                    elementSize = ClusterElementSize(element->type, element->isRepeated);
+                    // This should be the alignmentRequirement of the LV1DArray struct which should be 4 (contains int32 and int8)
```

Comment:
This change was in the preliminary PR where I had misunderstood how the LV1DArray would be stored in memory. I have since updated the changes so that we use the correct element size and alignment requirement.